src/infra/logging/logger.js

```diff
@@ -0,0 +1,113 @@
+const IO = require('crocks/IO');
+const getProp = require('crocks/Maybe/getProp');
+const isObject = require('crocks/predicates/isObject');
+const safe = require('crocks/Maybe/safe');
+const composeK = require('crocks/helpers/composeK');
+const maybeToArray = require('crocks/Maybe/maybeToArray');
+const chain = require('crocks/pointfree/chain');
+const objOf = require('crocks/helpers/objOf');
+
+const { Just } = require('crocks/Maybe');
+
+const {
+  createLogger,
+  format,
+  transports,
+} = require('winston');
+
+const {
+  combine,
+  timestamp,
+  label,
+  printf,
+  json,
+  colorize,
+} = format;
+
+const productionFormatter = (machineHost) => combine(
+  label({
+    label: machineHost,
+  }),
+  timestamp(),
+  json(),
+);
+
+const productionLogger = ({
+  machineHost,
+  slackWebHookChannel,
+  slackWebHookUrl,
+}) => createLogger({
+  format: productionFormatter(machineHost),
+  transports: [
+    new transports.Console(),
+    new transports.Slack({
+      level: 'warn',
+      channel: slackWebHookChannel,
+      webhook_url: slackWebHookUrl,
+      custom_formatter: (level, message, meta) => ({
+        text: `*${process.env.npm_package_name}*\n`
+          + `*${level.toUpperCase()}*: ${message}\n`
+          + `*TIME*: ${new Date()}\n`
+          + `*STACK*: ${meta.stack || '--'}`,
+      }),
+    }),
```

Comment:
I would actually not include this transport, as incoming webhooks are limited to a single request per second and there is not logging aggregation at the application level;